From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cheng Zhao <zcbenz@gmail.com>
Date: Sun, 14 Apr 2013 15:54:46 +0800
Subject: make_module_globalpaths_a_reference.patch

This PR makes Module.globalPaths a reference.

We need to hack the search paths of the require function so we can
load libraries from embedded applications without modifications of
node's module code.

(cherry picked from commit 76ba048c37588ee32636817fa7b8dffc64330cbf)

diff --git a/lib/internal/modules/cjs/loader.js b/lib/internal/modules/cjs/loader.js
index b09e3cceb55120d77e6e86b3425c3d7d447f53ad..aaf483abc0df336f9a1ea0a59deadaa12522c547 100644
--- a/lib/internal/modules/cjs/loader.js
+++ b/lib/internal/modules/cjs/loader.js
@@ -1076,8 +1076,8 @@ Module._initPaths = function() {
 
   modulePaths = paths;
 
-  // Clone as a shallow copy, for introspection.
-  Module.globalPaths = modulePaths.slice(0);
+  // clone as a shallow copy, for introspection.
+  Module.globalPaths = modulePaths;
 };
 
 Module._preloadModules = function(requests) {
